doc/sphinxman/source/build_planning.rst

```diff
@@ -0,0 +1,1152 @@
+
+    >>> cmake
+
+B. Build with specific Python
+
+  .. code-block:: bash
+
+    >>> cmake -DPYTHON_EXECUTABLE=/path/to/interp/python2.7
+
+C. Build with full Python specification to root directory ``${PFXC}``
+
+  .. code-block:: bash
+
+    >>> cmake -DPYTHON_EXECUTABLE="${PFXC}/bin/python" \
+              -DPYTHON_LIBRARY="${PFXC}/lib/libpython3.5m.so" \
+              -DPYTHON_INCLUDE_DIR="${PFXC}/include/python3.5m"
+
+
+.. _`faq:runtimepython`:
+
+What Python is Psi4 running
+---------------------------
+
+The Python detected at build-time is embedded into the |PSIfour|
+executable. That is, the top line of ``bin/psi4`` is something like
+``#!/path/to/miniconda/envs/p4deps/bin/python3.5``, and that's the Python
+through which |PSIfour| is running, *not* the Python of ``which python``.
+To use a different Python with |PSIfour| in the short term, just
+``path/to/desired/python psi4`` on the command line to override the
+shebang line. To use a different Python with |PSIfour| in the long term,
+edit the shebang line.
+
+If you're using |PSIfour| as a Python module, then |PSIfour| *is* running
+the Python of ``which python``.
+
+
+.. _`faq:wrongpyfalse`:
+
+How to fix "``undefined symbol: _Py_FalseStruct``"
+--------------------------------------------------
+
+You're probably loading a Py3-compiled Psi4 in Py2. Switch interpreters
+and re-run. A python of proper Py2 or Py3-ness is baked into the |PSIfour|
+"executable", so you'll see this error only for Psi4 as Python module.
+
+
+.. _`faq:gdblldb`:
+
+How to use ``gdb`` and ``lldb`` with Psi4
+-----------------------------------------
+
+Debugging |PSIfour| has gotten a little confusing now that it's running through Python. Here's the syntax ::
+
+  >>> cd {objdir}
+  >>> lldb -- python stage/{prefix}/bin/psi4 ../tests/tu1-h2o-energy/input.dat
+  >>> (lldb) run
+
+::
+
+  >>> cd {objdir}
+  >>> gdb python
+  >>> (gdb) run stage/{prefix}/bin/psi4 ../tests/tu1-h2o-energy/input.dat
+
+
+.. .. _`faq:valgrindpsi`:
+.. 
+.. How to use ``valgrind`` with Psi4
+.. ---------------------------------
+.. 
+.. When you naively use Valgrind with Psi4, you're likely to get incomprehensible mess of garbage or it may just crash with a boost overflow error. This happens because the boost python layer looks really really bad as far as Valgrind is concerned, i.e. it looks like a ton of memory leaks.  It really isn't, so we want to ignore all such errors/warnings. Valgrind has a mechanism for this in the way of suppression files.  Calling Valgrind as:
+.. 
+.. ```bash
+.. valgrind --suppressions=<file_name>
+.. ```
+.. 
+.. will run valgrind with the suppression file located on disk at "file_name". Lucky for you, Psi4 comes with such a suppression file at [``$top-level-psi4-dir/lib/valgrind-python.supp``](../blob/master/lib/valgrind-python.supp). This should remove all the boost python errors.
+..  
+.. The other error, boost overflow error arises from ``src/lib/libmints/sieve.cc`` where the inverse
```

Comment:
No, they're not. But the next time I hear of someone running valgrind, I'm going to quiz them on any procedural changes, then revive this section (commented now) with accurate info. The Boost bit is dead, but the suppression file and possibly something about sieving may endure.